client/src/Stores/MicrosoftLearn.store.ts

```diff
@@ -91,15 +99,85 @@ export class MicrosoftLearnStore extends ChildStore {
     const levels = toMap(catalog.levels, item => item.id);
     const allItems = [...modules, ...learningPaths];
     const items = toMap(allItems, item => item.uid);
+
     this.catalog = { contents: items, products, roles, levels };
     this.filteredCatalogContent = allItems;
     this.isLoadingCatalog = false;
+    this.productMap = this.getProductHierarchicalMap();
+
+    this.filter.catalog = this.catalog;
+    this.filter.productMap = this.productMap;
+    this.filter.setDisplayFilters(this.filteredCatalogContent, false);
+
+    this.loadFiltersFromPath();
+  }
+
+  @action
+  addFilter(type: FilterType, filters: string[]){

```

Comment:
Thank you for this comment. I think this would be a great way to structure. Implementationally, correct me if I am wrong- This can be done by having the `FilterStore` simply `extend` the existing `LearnStore`, and then using the `FilterStore` everywhere in the filter components. The user actions would trigger changes to the filter observables, which would internally change the `LearnStore` states.

Reply:
I was just exploring ways to achieve this, and I think that since we are moving the `searchTerm` and everything else related to filtering to a separate store, probably can do the same to `filteredCatalogContent` as well? This will substantially reduce the dependency on `learnStore`, where we would only be handling the selected catalog items. Everything on the filtered content inside the `FilterStore`.

Reply:
I have added a separate store for filtering in my latest commit. In the future, in case a new API for filtering is created, the `actions` can be modified accordingly to make calls to the backend. With this new store, the segregation now is such that:

- `MicrosoftLearnStore`: makes the initial call to the catalog-api; caters to the selection and removal of content items for assignments.
- `MicrosoftLearnFilterStore`: all the filtering logic is encapsulated here; have moved the `filteredCatalogContent` to this store for ease in updating the same.